new functionality and compatibility

nadavbh12 opened the pull request:
See the notes for the second commit.
It requires minor modifications to the ResultsLog constructor, otherwise should be the same with added functionality.

nadavbh12 committed:
added set_global_seeds

nadavbh12 committed:
new setup function, functionality, doc, resume

-added setup_logging_and_results
-export args to json function
-add resume functionality
-support latest bokeh
-add some documentation
-save to json rather than csv (much faster)
-add html title
-handle single y in plot

nadavbh12 committed:
fix multiple logger handlers

eladhoffer commented on log.py:
```diff
@@ -28,45 +77,106 @@ def setup_logging(log_file='log.txt'):
+        """
+        os.makedirs(path, exist_ok=True)
+        full_path = os.path.join(path, 'results')
+        self.date_path = '{}.json'.format(full_path)
```
Can you make json vs csv optional?

eladhoffer commented on misc.py:
```diff
@@ -23,3 +25,16 @@ def onehot(indexes, N=None, ignore_index=None):
+        pass
+    else:
+        torch.manual_seed(i)
+        if cuda:
```
you can use torch.cuda.is_available()

nadavbh12 committed:
json/csv, cuda check, default args

-make json/csv optional
-set default args in case they aren't defined in args
-check for cuda inside set_global_seeds

eladhoffer reviewed (approved):
Thanks!
